src/main/scala/com/fulcrumgenomics/util/GeneAnnotations.scala

```diff
@@ -25,32 +25,92 @@
 
-  /** A gene with zero or more transcripts, that all are on the same transcription strand. */
-  case class Gene(contig: String, start: Int, end: Int, negativeStrand: Boolean, name: String, transcripts: Seq[Transcript])
-    extends Interval(contig, start, end, negativeStrand, name) with Iterable[Transcript] {
-    def iterator: Iterator[Transcript] = transcripts.iterator
+    /** Returns the locus with the maximum transcript mappings. */
+    def primaryLocus: GeneLocus = loci.maxBy(_.size)
   }
 
-  /** A transcript associated with a given gene (which stores the transcription strand).  Contains zero or more exons.
-    * The exons should be given in transcript order. */
-  case class Transcript(name: String, start: Int, end: Int, cdsStart: Int, cdsEnd: Int, exons: Seq[Exon]) {
+
+  /**
+    * A gene locus representing a set of transcripts that are mapped to the same location on the genome.
+    *
+    * @param transcripts the set of transcripts at the locus
+    */
+  case class GeneLocus(transcripts: Seq[Transcript]) extends Locatable with Iterable[Transcript] {
+    require(transcripts.nonEmpty, "GeneLocus cannot be generated from an empty set of transcripts.")
+    require(transcripts.forall(t => t.chrom == transcripts.head.chrom))
+    require(transcripts.forall(t => t.negativeStrand == transcripts.head.negativeStrand))
+
+    val chrom: String = transcripts.head.chrom
+    val start: Int = transcripts.minBy(_.start).start
+    val end:   Int = transcripts.maxBy(_.end).end
+    val negativeStrand: Boolean = transcripts.head.negativeStrand
+    def positiveStrand: Boolean = !negativeStrand
+
+    // Locatable implementation
+    override def getContig: String = chrom
+    override def getStart: Int = start
+    override def getEnd: Int = end
+
+    override def iterator: Iterator[Transcript] = this.transcripts.iterator
+  }
+
+
+  /**
+    * A transcript [mapping] associated with a given gene locus.  Contains zero or more exons.
+    * The exons should be given in the order of transcription.
+    */
+  case class Transcript(name: String,
+                        chrom: String,
+                        start: Int,
+                        end: Int,
+                        cdsStart: Option[Int],
+                        cdsEnd: Option[Int],
+                        negativeStrand: Boolean,
+                        exons: Seq[Exon]) extends Locatable {
     if (exons.length > 1) {
       val exonsOverlap = genomicOrder.sliding(2).map { e => (e.head, e.last) }.exists { case (e1, e2) => CoordMath.overlaps(e1.start, e1.end, e2.start, e2.end) }
       require(!exonsOverlap, s"exons overlap for transcript: $name")
     }
+
+    require(cdsStart.isDefined == cdsEnd.isDefined, s"cdsStart and cdsEnd must either both or neither be defined on tx: $name.")
+
+    // Locatable implementation
+    override def getContig: String = chrom
+    override def getStart: Int = start
+    override def getEnd: Int = end
```

Comment:
This should just be a trait somewhere that we can mix in, and then we can standardize on the real names for these.

Reply:
I agree .. and I think I've done that elsewhere.  I'll dig it up and open up a separate PR in the near future.